ros2topic/ros2topic/verb/info.py

```diff
@@ -12,11 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from rclpy.qos import QoSDurabilityPolicy
+from rclpy.qos import QoSLivelinessPolicy
+from rclpy.qos import QoSReliabilityPolicy
+
 from ros2cli.node.direct import DirectNode
 from ros2topic.api import get_topic_names_and_types
 from ros2topic.api import TopicNameCompleter
 from ros2topic.verb import VerbExtension
+import logging
+logger = logging.getLogger(__name__)
 
+def print_topic_info(topic_name, get_topic_info_func):
+    try:
+        for info in get_topic_info_func(topic_name):
+            print('\nNode Name: %s' % info['node_name'])
+            print('Node Namespace: %s' % info['node_namespace'])
+            print('Topic Type: %s' % info['topic_type'])
+            print('GID: %s' % '.'.join(format(x, '02x') for x in info['gid']))
+            print('QoS Profile:')
+            qos_profile = info['qos_profile']
+            print('  Reliability: %s' % QoSReliabilityPolicy(qos_profile['reliability']).name)
+            print('  Durability: %s' % QoSDurabilityPolicy(qos_profile['durability']).name)
+            print('  Lifespan: %d nanoseconds' % qos_profile['lifespan'].nanoseconds)
+            print('  Deadline: %d nanoseconds' % qos_profile['deadline'].nanoseconds)
+            print('  Liveliness: %s' % QoSLivelinessPolicy(qos_profile['liveliness']).name)
+            print('  Liveliness Lease Duration: %d nanoseconds' %
+                  qos_profile['liveliness_lease_duration'].nanoseconds)
+    except NotImplementedError as e:
+        logger.warning(str(e))
+    except Exception as e:
+        logger.error(str(e))
```

Comment:
@jaisontj @mm318 consider:

- Not catching all but only the ones you'd expect.
- Not using the `logging` module but returning the error string from `InfoVerb.main` instead.

Reply:
Done.